src/zope/sqlalchemy/datamanager.py

```diff
@@ -84,22 +85,18 @@ def __init__(
             self, session, status, transaction_manager, keep_session=False):
         self.transaction_manager = transaction_manager
 
-        # Support SQLAlchemy 2.0
-        # https://docs.sqlalchemy.org/en/14/orm/session_api.html?highlight=get_transaction#sqlalchemy.orm.Session.get_transaction
-        transaction = (
-            session.get_transaction()
-            if hasattr(session.transaction, "get_transaction")
-            else session.transaction
-        )
-
-        # Support both SQLAlchemy 1.0 and 1.1
-        # https://github.com/zopefoundation/zope.sqlalchemy/issues/15
-        _iterate_parents = (
-            getattr(transaction, "_iterate_self_and_parents", None)
-            or transaction._iterate_parents
-        )
-
-        self.tx = _iterate_parents()[-1]
+        if sqlalchemy_version >= "1.4.0":
```

Comment:
This will break at version `1.10.0` as:

```python
>>> '1.10.0' >= '1.4.0'
False
```

But it is good enough for now. We will drop support for versions 1.0 and 1.1 before SQLAlchemy will reach that version number.

Reply:
I think this should not be necessary. The comment was mostly a note to myself to be careful in other projects.